ProGetAutomation/Functions/Publish-ProGetUniversalPackage.ps1

```diff
@@ -202,9 +203,18 @@ See http://inedo.com/support/documentation/various/universal-packages/universal-
             $packageStream = New-Object 'IO.FileStream' ($PackagePath, 'Open', 'Read')
             $streamContent = New-Object 'Net.Http.StreamContent' ([IO.Stream]$packageStream)
             $streamContent.Headers.ContentType = New-Object 'Net.Http.Headers.MediaTypeHeaderValue' ('application/octet-stream')
-            $httpResponseMessage = $httpClient.PutAsync($proGetPackageUri, [Net.Http.HttpContent]$streamContent)
+            $canceller = New-Object 'Threading.CancellationTokenSource'
+            $httpResponseMessage = $httpClient.PutAsync($proGetPackageUri, [Net.Http.HttpContent]$streamContent, $canceller.Token)
             if( -not $httpResponseMessage.Wait($maxDuration) )
             {
+                $canceller.Cancel()
+                $maxTries = 1000
+                $tryNum = 0
+                while( $tryNum -lt $maxTries -and -not $httpResponseMessage.IsCanceled )
+                {
+                    $tryNum += 1
+                    Start-Sleep -Milliseconds 100

```

Comment:
The error message uses the value of $Timeout, but this block is coded for 1000 tries of 100ms. Can we adjust so these remain in sync?

Reply:
This is waiting for the cancel to complete, not the upload. The `$Timeout` has already been exceeded. In my testing, the cancel only takes less than a second, but if we don't wait for it, later on, when we go to dispose all the disposable objects we've created, sometimes disposing the upload task fails because the cancel hasn't finished yet,